make/mt-os/mt-linux.mk

```diff
@@ -35,4 +35,8 @@ IOTC_COMMON_COMPILER_FLAGS += -Wno-format
 
 IOTC_CONFIG_FLAGS += -DIOTC_MULTI_LEVEL_DIRECTORY_STRUCTURE
 IOTC_CONFIG_FLAGS += -DIOTC_LIBCRYPTO_AVAILABLE
+
+# need for x86 wolfssl builds
+IOTC_CONFIG_FLAGS += -DWOLFSSL_X86_64_BUILD
```

Comment:
It's a first for us to put TLS library specifics in the OS make target file as there's supposed to be a delineation between the two.  Still, I can't figure a better way to do this without making some larger changes to our mt file topography.

Could you please surround it with a check to see what the current bsp tls configuration is so that -DWOLFSSL defines don't appear in the command line arguments for non WolfSSL (ie: mbedTLS) builds?

For instance:

```
ifneq (,$(findstring wolfssl,$(IOTC_TLS_LIB_NAME)))
  # need for x86 wolfssl builds
  IOTC_CONFIG_FLAGS += -DWOLFSSL_X86_64_BUILD
endif

```

Reply:
Yes, I do understand the struggle, but quite plainly, the old system was crippling our build. So while it shouldn't be here, it DEFINITELY shouldn't be in the global `wolfssl` config. One option you could consider moving forward (for `wolfssl` specifically) is to provide guidelines for using the `wolfssl` `user_settings.h` file for end developers to specify build optimizations like `USE_FAST_MATH` and drop them from your autoconfigure. I feel relatively confident that a set of flags that would avoid platform specific issues to could be added to the makefile and then it would be up to the end user to add in `TFM` related optimizations. Perhaps something we could do in a separate PR?

https://www.wolfssl.com/docs/frequently-asked-questions-faq/#How_do_I_manage_the_build_configuration_for_wolfSSL?